### ISSUE TITLE
[FEATURE] +value mixin called with mixin

### ISSUE THREAD

**Issue by MathiasWinckel** [RosaeNLG/rosaenlg#103]
**Is your feature request related to a problem? Please describe.**

I would love to use the "value" mixin with mixin

Let's try to evaluate the following template:
```
//- sub mixin

mixin passedMixin()
    | print this.

//- main mixin

mixin firstTopMixin(oneMixin)
    //- value of sub template
    +value(oneMixin)

//- print top mixin

+firstTopMixin(passedMixin)
```

It tries a `+value` call on a mixin, directly passed as an argument.
Of course, the idea is that the sub mixin could be from another files, stored as an attribute of objects, etc...

I would expect this to generate "Print this." but in the current official version it fails saying:
`value not possible on: undefined`
because nothing is done to detect mixin and evaluate them, and for some reason it prints the template object as undefined...

**Describe the solution you'd like**

A way to allow +value to use a mixin as first parameter, to allow more powerfull template writing

I confess I did not look for the cleaner and meaningful way to do that, but the engine could do it quite simply in the ValueManager with a simple test on the `firstParam` for wrapping a mixin as a ref and keep going. 
As far as I can tell, it works. But I could keep going with my testing before making a PR.

**Describe alternatives you've considered**

Actually, you can already quite simply pass the mixin as the ref of a wrapping object:
`+firstTopMixin({ ref: passedMixin})`
and I guess I would use this trick if the engine stays as it is. But the engine could do that for you, and I thought it could be an interesting feature to handle mixin.

**Additional context**

Handling parameters for the passed mixin could be very tricky, and I think it would need a far more complex solution.
But there may be a quite simple trick to achieve it writing my template, using a wrapping mixin:
```
//- sub mixin

mixin passedMixin(arg1)
    | #{arg1}
    
//- main mixins

mixin firstTopMixin(text1)
    //- mixin wrapping possible arg passing
    mixin wrappingMixin()
        +passedMixin(text1)

    //- value of sub template
    +value(wrappingMixin)

//- print top mixins
+firstTopMixin("print that.")
```
should output "Print that."

here I use a `wrappingMixin` when I have the parameter available for my `passedMixin`, inside my `firstTopMixin` and simply pass the wrapper to +value
It's a bit messier, to have wrapping mixin defined in the middle, but quite powerfull for such a simple trick...

PS: I know I already have some issues opened, sometimes with PRs waiting... Sorry about that, I'll keep up with all of that in January.

**Comment by ludans** [comment]
I understand that you want to dynamically call a mixin, knowing its name at runtime.
All mixins are available through the map `pug_mixins`. I references plain JavaScript functions (with parameters, which are the parameters of the mixin).

```
//- sub mixin
mixin passedMixin(param)
  | print this #{param}.

//- main mixin

mixin firstTopMixin(oneMixin, someParam)
  //- value of sub template
  - pug_mixins[oneMixin](someParam)

//- print top mixin
+firstTopMixin('passedMixin', 5)
```

https://rosaenlg.org/rosaenlg/3.0.3/mixins_ref/otherAndHelpers.html#_pug_mixins_map

will generate: `Print this 5.`

Tell me if it solves your issue.

**Comment by MathiasWinckel** [comment]
I would not mind passing mixin name instead of mixin, but what bother me is that it relies on handling your content differently if it's a mixin or let say a string.

When having your `value` mixin handling the mixins allows one to do more generic templates. 
Maybe it's not the cleaner but I tend to look for a +value print function handling more content because of that.

Maybe a less abstract example would be:
your top mixin is building the structure around titles and you want to pass the title content, but it could be a string or a mixin, and you don't want to deal with that in your title printer:
```
//- sub mixin

mixin title2()
    |  title 2
    
//- main mixins

mixin printTitle(textContent)
    h1
        +value(textContent)

//- print titles
div
    +printTitle("title 1")
div
    +printTitle(title2)
```
it would generate `<div><h1>Title 1</h1></div><div><h1>Title 2</h1></div>`
and you were able to have a generic printTitle template as long as +value can handle the `textContent`.

As far I can tell, it seems I can currently do that really easily with the ref attribute trick:
`+printTitle({ ref: title2 })`
I only need to provide a "working" ref attribute to what I want to pass to +value to achieve what I want, that's not the end of the world. 
But maybe it could be interesting if the `value` mixin would natively handle mixins ?

**Comment by ludans** [comment]
I implemented it in v3.0.4: https://rosaenlg.org/rosaenlg/3.0.4/mixins_ref/value.html#_value_of_a_mixin

Can you please make a test?
